container/shim/src/index.js

```diff
@@ -42,161 +47,231 @@ const PROXY_RESPONSE_HEADERS = [
     }
-    for (const key of ['filename', 'download']) {
-      if (key in req.query) {
-        ipfsUrl.searchParams.set(key, req.query[key])
-      }
+  }
+
+  const controller = new AbortController()
+  const timeout = setTimeout(() => {
+    controller.abort()
+  }, GATEWAY_TIMEOUT)
+
+  const ipfsReq = https.get(ipfsUrl, {
+    agent: ipfsAgent,
+    timeout: GATEWAY_TIMEOUT,
+    headers: { 'User-Agent': NODE_UA },
+    signal: controller.signal
+  }, async fetchRes => {
+    clearTimeout(timeout)
+    const { statusCode } = fetchRes
+    if (statusCode >= 400) {
+      debug.extend('error')(`Invalid response from IPFS gateway (${statusCode}) for ${cid}`)
     }
 
-    const controller = new AbortController()
-    const timeout = setTimeout(() => {
-      controller.abort()
-    }, GATEWAY_TIMEOUT)
-
-    const ipfsReq = https.get(ipfsUrl, {
-      agent: ipfsAgent,
-      timeout: GATEWAY_TIMEOUT,
-      headers: { 'User-Agent': NODE_UA },
-      signal: controller.signal
-    }, async fetchRes => {
-      clearTimeout(timeout)
-      const { statusCode } = fetchRes
-      if (statusCode >= 400) {
-        debug.extend('error')(`Invalid response from IPFS gateway (${statusCode}) for ${cid}`)
-      }
+    res.status(statusCode)
+    proxyResponseHeaders(fetchRes, res)
 
-      res.status(statusCode)
-      proxyResponseHeaders(fetchRes, res)
+    if (format === 'car') {
+      streamCAR(fetchRes, res).catch(() => {})
+    } else {
+      fetchRes.pipe(res)
+    }
+  }).on('error', err => {
+    clearTimeout(timeout)
+    debug.extend('error')(`Error fetching from IPFS gateway for ${cid}: ${err.name} ${err.message}`)
+    if (controller.signal.aborted) {
+      return res.sendStatus(504)
+    }
+    res.sendStatus(502)
+  }).on('timeout', () => {
+    clearTimeout(timeout)
+    debug.extend('error')(`Timeout from IPFS gateway for ${cid}`)
+    ipfsReq.destroy()
+    res.destroy()
+  })
 
-      if (format === 'car') {
-        streamCAR(fetchRes, res).catch(() => {})
-      } else {
-        fetchRes.pipe(res)
-      }
-    }).on('error', err => {
-      clearTimeout(timeout)
-      debug.extend('error')(`Error fetching from IPFS gateway for ${cid}: ${err.name} ${err.message}`)
-      if (controller.signal.aborted) {
-        return res.sendStatus(504)
-      }
-      res.sendStatus(502)
-    }).on('timeout', () => {
-      clearTimeout(timeout)
-      debug.extend('error')(`Timeout from IPFS gateway for ${cid}`)
+  req.on('close', () => {
+    clearTimeout(timeout)
+    if (!res.writableEnded) {
+      debug.extend('error')('Client aborted early, terminating gateway request')
       ipfsReq.destroy()
-      res.destroy()
-    })
+    }
+  })
+}
 
-    req.on('close', () => {
-      clearTimeout(timeout)
-      if (!res.writableEnded) {
-        debug.extend('error')('Client aborted early, terminating gateway request')
-        ipfsReq.destroy()
-      }
-    })
+app.get('/register/:l2NodeId', async function (req, res) {
```

Comment:
Ditto, express does not support async functions for request handlers.

Reply:
https://github.com/filecoin-saturn/L1-node/pull/48